core/src/main/java/com/yahoo/imapnio/async/data/StoreResult.java

```diff
@@ -0,0 +1,99 @@
+package com.yahoo.imapnio.async.data;
+
+import java.util.List;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.sun.mail.imap.protocol.IMAPResponse;
+
+/**
+ * This class provides the highest modification sequence, the list of IMAP responses, and the list of modified messages number
+ * from store command response based on RFC3501 and RFC7162 as following.
+ *
+ * <pre>
+ * resp-text-code      =/ "HIGHESTMODSEQ" SP mod-sequence-value /
+ *                        "MODIFIED" SP sequence-set
+ * </pre>
+ */
+public class StoreResult {
+
+    /** The highest modification sequence. */
+    private Long highestModSeq;
+
+    /** The collection of IMAP responses. */
+    private final List<IMAPResponse> imapResponses;
+
+    /** The collection of message number as sequence-set, only shown when CondStore is enabled. */
+    private final MessageNumberSet[] modifiedMsgSets;
+
+    /**
+     * Initializes a {@code StoreResult} object with IMAP responses collection.
+     *
+     * @param imapResponses collection of IMAP responses from store command result
+     */
+    public StoreResult(@Nonnull final List<IMAPResponse> imapResponses) {
```

Comment:
in what use case you need this constructor and line 47? I only saw line 57 getting used.